### ISSUE TITLE
[Unity 2021] Local build has issues

### ISSUE THREAD

**Issue by Kinerius** [decentraland/unity-renderer#2709]
Check the issue here: https://github.com/decentraland/unity-renderer/pull/2602#issuecomment-1190714143

![image.png](https://images.zenhubusercontent.com/61080c59fd1a374ce3f3b027/fac13572-c493-4046-8470-966d1e8a4e9e)

This is a problem because:

- We won't be able to debug stuff using local builds
- preview-mode runs on localhost, so creators that run their scene locally when working on it, probably will have the same problem

**Comment by Antivortex** [comment]
More detailed errors:

![image.png](https://images.zenhubusercontent.com/62fb937fa41da238b0ed7951/e83d5afc-0471-404e-aecc-32ba78d93e7b)

**Comment by Antivortex** [comment]
This is an HTTP error 500, happens only in Unity 2021 WebGL build, so far not clear why:

![image.png](https://images.zenhubusercontent.com/62fb937fa41da238b0ed7951/d7c73259-8c5c-401c-941e-52360f781e12)

Some information on this topic:

https://forum.unity.com/threads/http-1-1-500-internal-server-error-while-calling-a-web-service-by-wwwform.161483/

**Comment by Antivortex** [comment]
After close investigation this looks like Unity internal bug which might be fixed by some unknown configuration or might be not. Here are some details:

1) Error code is 500
2) WebGL client is using UnityWebRequest to fetch a file from localhost.
3) After request fails webRequest.result field has value Connection Error, webRequest.error filed has value Unknown Error
4) Example of URL used: https://127.0.0.1:8000/@/artifacts/...mQyeu5ZjFzJFZ2bz4zdYX8yjRFS1QqCg4CF67XuQc31Uw
5) This http error does not appear in the Chrome's Developer tools Network tab.
6) This error does not happen if the build made with Unity 2020.3.34f1 with the same project and Javascript server running, for Unity 2020 WebGL client gets correct response and downloads the file.
7) If URL is opened in the browser, the file downloads without any issues, server runs without issues.


A thread on Unity forum was created:

https://forum.unity.com/threads/http-error-500-in-webgl-build-while-fetching-localhost-url-using-unity-2021-unitywebrequest.1343882/

**Comment by Antivortex** [comment]
After a call with Mateo we figured that if you run local server and provide a renderer branch in the URL parameters to download Unity 2021 CI build it works well, but when we do local copy-pasting of the WebGL binaries this error starts happening for 2021 builds. We always did a copy pasting like this in the past for 2020 builds and it works, but with 2021 something changed. Most likely it is related to the changes in JS code for WebGL builds (we only copied binaries).
